tests/HopSpec/AnfSpec.hs

```diff
@@ -0,0 +1,340 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module HopSpec.AnfSpec
+  ( spec
+  ) where
+
+import Hopper.Internal.LoweredCore.ANF
+import Hopper.Internal.Core.Term
+import Hopper.Internal.Core.Literal
+import Hopper.Internal.Type.Relevance
+import Hopper.Utils.LocallyNameless
+
+import Test.Hspec
+import Test.Hspec.Expectations
+import qualified Data.Text as T
+import qualified Data.Vector as V
+
+spec :: Spec
+spec =
+  describe "ANF" $ do
+    describe "toAnf" $ do
+      let v0 = LocalVar $ LocalNamelessVar 0 $ BinderSlot 0
+          v0_0 = v0
+          v0_1 = LocalVar $ LocalNamelessVar 0 $ BinderSlot 1
+          v1 = LocalVar $ LocalNamelessVar 1 $ BinderSlot 0
+          v2 = LocalVar $ LocalNamelessVar 2 $ BinderSlot 0
+          add = GlobalVarSym $ GlobalSymbol $ T.pack "add"
```

Comment:
If only one non string overloaded string instance is in scope, I'm ok with overloaded strings. Otherwise qualified import T.pack, Bs.pack et al are a style I prefer to mark what the expected representation target is.  I kinda always worry that overloaded strings is one of those pragmas that make it easy to write hard to read / understand code personally.